### DISCUSSION TITLE
Using UpDownCounter in NET6: possible?

### DISCUSSION THREAD

**Discussion by julealgon** [open-telemetry/opentelemetry-dotnet#3256] [Q&A]
I have a scenario that requires me to post positive and negative deltas in the same counters. I noticed that the `UpDownCounter` was introduced for this reason here:
- https://github.com/dotnet/runtime/pull/64365

From what I understand, the standard `Counter<T>` is supposed to be monotonically increasing, so it shouldn't be used when there is a need to report both increments _and_ decrements, and `UpDownCounter<T>` should be used instead.

Since `UpDownCounter<T>` will only come as part of .NET 7, is there a way to temporarily add support for it in a .NET 6 codebase? In the past, it was possible to add support for some newer attributes etc by adding those manually to a codebase using the correct namespaces and they would be picked up automatically as long as you were using the latest compiler. Is something like that feasible regarding `UpDownCounter<T>`? Will the counter "just work" if I copy the code from that NET 7 PR into my project and keep the same `System` namespaces in place?

If that doesn't work, is there an alternative today to allow for a counter that pushes both increments and decrements to be reported to OpenTelemetry as a proper "up/down counter" (i.e. one that is not monotonic)?

**Comment by cijothomas** [comment]
> Will the counter "just work"

Unfortunately, no. The OpenTelemetry SDK does not understand UpDownCounters yet.

There is no real alternative but to wait for this SDK to start supporting it. It'll be stable released on Nov 2022, with several non-stable ones coming months before that. (You may use Gauge, but not really a workaround....)

**Reply by julealgon** [reply]
> The OpenTelemetry SDK does not understand UpDownCounters yet.

Oh, not even the preview versions do?

> The OpenTelemetry SDK does not understand UpDownCounters yet.

What if I just want to hack it in? It's supported by the OpenTelemetry collector, right? What would one need to change to "convert" a normal counter into an updown one? Is it just the type + the monotonic flag, or is there more to it than that?

Also, what is the consequence of using a standard `Counter` today with both positive and negative deltas? From what I'm seeing here, no exception or error is raised, values are accumulated correctly and sent to both the console and my file-based test collector. Will something not work properly somewhere if I keep using it that way until the official `UpDownCounter` is introduced?
